MSAL/test/app/mac/MSALAcquireTokenViewController.m

```diff
@@ -0,0 +1,366 @@
+    else
+    {
+        [self.scopesLabel setStringValue:defaultScope];
+        self.selectedScopes = @[defaultScope];
+    }
+}
+
+- (void)updateResultView:(MSALResult *)result
+{
+    NSString *resultText = [NSString stringWithFormat:@"{\n\taccessToken = %@\n\texpiresOn = %@\n\ttenantId = %@\n\tuser = %@\n\tscopes = %@\n\tauthority = %@\n}",
+                            [result.accessToken msidTokenHash], result.expiresOn, result.tenantId, result.account, result.scopes, result.authority];
+    
+    [self.resultView setString:resultText];
+    
+    NSLog(@"%@", resultText);
+}
+
+- (void)updateResultViewError:(NSError *)error
+{
+    NSString *resultText = [NSString stringWithFormat:@"%@", error];
+    [self.resultView setString:resultText];
+    NSLog(@"%@", resultText);
+}
+
+- (MSALPromptType)promptType
+{
+    NSString *promptType = [self.promptBehavior labelForSegment:[self.promptBehavior selectedSegment]];
+    
+    if ([promptType isEqualToString:@"Select"])
+        return MSALPromptTypeSelectAccount;
+    if ([promptType isEqualToString:@"Login"])
+        return MSALPromptTypeLogin;
+    if ([promptType isEqualToString:@"Consent"])
+        return MSALPromptTypeConsent;
+    
+    @throw @"Do not recognize prompt behavior";
+}
+
+- (BOOL)passedInWebview
+{
+    NSString* webViewType = [self.webViewType labelForSegment:[self.webViewType selectedSegment]];
+    
+    if ([webViewType isEqualToString:@"MSAL"])
+    {
+        return NO;
+    }
+    else if ([webViewType isEqualToString:@"Passed In"])
+    {
+        return YES;
+    }
+    else
+    {
+        @throw @"unexpected webview type";
+    }
+}
+
+- (void)showAlert:(NSString *)messageText informativeText:(NSString *)informativeText
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        NSAlert *alert = [[NSAlert alloc] init];
+        alert.messageText = messageText;
+        [alert addButtonWithTitle:@"OK"];
+        alert.informativeText = informativeText;
+        [alert runModal];
+    });
+}
+
+- (IBAction)clearCache:(id)sender
+{
+    MSALTestAppSettings *settings = [MSALTestAppSettings settings];
+    
+    // Delete accounts.
+    NSDictionary *currentProfile = [MSALTestAppSettings currentProfile];
+    NSString *clientId = [currentProfile objectForKey:MSAL_APP_CLIENT_ID];
+    NSString *redirectUri = [currentProfile objectForKey:MSAL_APP_REDIRECT_URI];
+    __auto_type authority = [settings authority];
+    
+    NSError *error = nil;
+    MSALPublicClientApplication *application = [[MSALPublicClientApplication alloc] initWithClientId:clientId
+                                                                                           authority:authority
+                                                                                         redirectUri:redirectUri
+                                                                                               error:&error];
+    
+    BOOL result = [application.tokenCache clearWithContext:nil error:&error];
```

Comment:
nit: do you need to create public client application here?

Reply:
Will update cache related code in the next PR.